hoomd/pytest/test_gsd.py

```diff
@@ -0,0 +1,378 @@
+    integrator.forces.append(lj)
+    langevin = hoomd.md.methods.Langevin(hoomd.filter.All(), kT=1, seed=1)
+    integrator.methods.append(langevin)
+    sim.operations.integrator = integrator
+
+    gsd_writer = hoomd.write.GSD(filename=filename,
+                                 trigger=hoomd.trigger.Periodic(1),
+                                 filter=hoomd.filter.Type(['p2']),
+                                 mode='wb')
+    sim.operations.writers.append(gsd_writer)
+
+    sim.run(5)
+    snapshot = sim.state.snapshot
+
+    with gsd.hoomd.open(name=filename, mode='rb') as traj:
+        for step in range(5):
+            np.testing.assert_array_equal(traj[step].particles.typeid,
+                                          [1] * int(snapshot.particles.N / 2))
+
+
+def test_write_gsd_truncate(simulation_factory, device,
+                            hoomd_snapshot, tmp_path):
+
+    filename = tmp_path / "temporary_test_file.gsd"
+
+    sim = simulation_factory(hoomd_snapshot)
+
+    integrator = hoomd.md.Integrator(dt=0.005)
+    lj = hoomd.md.pair.LJ(nlist=hoomd.md.nlist.Cell(), r_cut=2.5)
+    lj.params[('p1', 'p1')] = {'sigma': 1, 'epsilon': 5e-200}
+    lj.params[('p1', 'p2')] = {'sigma': 1, 'epsilon': 5e-200}
+    lj.params[('p2', 'p2')] = {'sigma': 1, 'epsilon': 5e-200}
+    integrator.forces.append(lj)
+    langevin = hoomd.md.methods.Langevin(hoomd.filter.All(), kT=1, seed=1)
+    integrator.methods.append(langevin)
+    sim.operations.integrator = integrator
+
+    gsd_writer = hoomd.write.GSD(filename=filename,
+                                 trigger=hoomd.trigger.Periodic(1),
+                                 truncate=True,
+                                 mode='wb')
+    sim.operations.writers.append(gsd_writer)
+
+    snapshot_list = []
+    for _ in range(2):
+        sim.run(1)
+        snapshot_list.append(sim.state.snapshot)
+
+    with gsd.hoomd.open(name=filename, mode='rb') as traj:
+        assert_equivalent_snapshots(traj[0], snapshot_list[-1])
+        assert_equivalent_snapshots(traj[-1], snapshot_list[-1])
+
+
+def test_write_gsd_dynamic(simulation_factory, device,
+                           hoomd_snapshot, tmp_path):
+
+    filename = tmp_path / "temporary_test_file.gsd"
+
+    sim = simulation_factory(hoomd_snapshot)
+
+    integrator = hoomd.md.Integrator(dt=0.005)
+    lj = hoomd.md.pair.LJ(nlist=hoomd.md.nlist.Cell(), r_cut=2.5)
+    lj.params[('p1', 'p1')] = {'sigma': 1, 'epsilon': 5e-200}
+    lj.params[('p1', 'p2')] = {'sigma': 1, 'epsilon': 5e-200}
+    lj.params[('p2', 'p2')] = {'sigma': 1, 'epsilon': 5e-200}
+    integrator.forces.append(lj)
+    langevin = hoomd.md.methods.Langevin(hoomd.filter.All(), kT=1, seed=1)
+    integrator.methods.append(langevin)
+    sim.operations.integrator = integrator
+
+    gsd_writer = hoomd.write.GSD(filename=filename,
+                                 trigger=hoomd.trigger.Periodic(1),
+                                 mode='wb', dynamic=['momentum'])
```

Comment:
We could have a base `GSD` writer fixture since we use this exact code so much.

Reply:
Although these several lines are very similar in each test, we change different arguments almost every time we use it. We could define a function instead, but it would be similar to what we have now.

Reply:
I am not sure this is the case, we do that many times, but this is used fairly frequently. If you don't feel that that would improve clarity, then leave it as is, a function would obfuscate the code.

Reply:
The exact piece of code was used twice. At other places, we either used different ```mode``` or different ```filter``` etc. I'm not sure how to make a fixture that we can easily change the values passed to these arguments.

Reply:
Then that is fine. Leave it as is.